### PR TITLE
Add support for `ldpaths` detection on macOS via `DYLD_FALLBACK_LIBRARY_PATH`

### PR DESCRIPTION
**Pull request by mjsteinbaugh**
This proposed pull request parses the `etc/ldpaths` file in `R.home()` for `DYLD_FALLBACK_LIBRARY_PATH` on macOS.

I've confirmed that this works on my machine after testing with `python3 setup.py install`.

Best,
Mike

### CODE REVIEW

**Comment by mjsteinbaugh** [comment]
Attempts to address related issue https://github.com/randy3k/radian/issues/413

**Comment by mjsteinbaugh** [comment]
For reference, this patch is now applied in [koopa](https://koopa.acidgenomics.com/):
https://github.com/acidgenomics/koopa/blob/main/lang/shell/bash/include/install/common/shared/radian.sh

**Comment by randy3k** [comment]
As indicated in the ldpaths file.

```
## This is DYLD_FALLBACK_LIBRARY_PATH on Darwin (macOS) and
## LD_LIBRARY_PATH elsewhere.
## However, on macOS >=10.11 (if SIP is enabled, the default), the
## environment value will not be passed to a script such as R.sh, so
## would not seen here.
```

I thought that this trick no longer works for macOS >=10.11 with sip, apparently it is not the case?

**Comment by mjsteinbaugh** [comment]
@randy3k That patch works on my Mac running Ventura 13.5 with SIP enabled.

This is also a known issue with RStudio, which they're working to address in a new release:
https://github.com/rstudio/rstudio/issues/13085

**Comment by randy3k** [comment]
Oh, I think I have been misunderstanding the text in ldpaths for years. Actually it meant that we have to set the variable manually, the variable will not be passed automatically.

Thanks for raising it.

File: radian/app.py [hunk 1]
```diff
@@ -75,35 +75,35 @@ def main(cleanup=None):
     if not r_home:
         raise RuntimeError("Cannot find R binary. Expose it via the `PATH` variable.")
 
-    if sys.platform.startswith("linux"):
```
Comment (randy3k) [inline radian/app.py:78]: I think we should still restrict ourself to non-window.

For example 
```
 if not sys.platform.startswith("win"):
```

Comment (mjsteinbaugh) [reply radian/app.py:78]: Agreed yeah that makes sense

Comment (randy3k) [reply radian/app.py:78]: Do you want to do the edit?

Comment (mjsteinbaugh) [reply radian/app.py:78]: @randy3k Sure let me add that, 1 sec

**Review by randy3k — APPROVED** [review]

**Pull request merged**